WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -322,10 +323,14 @@ protected SendControlViewModel(Wallet wallet, string title)
 
 					await BuildTransaction(Password, intent, feeStrategy, allowUnconfirmed: true, allowedInputs: selectedCoinReferences);
 				}
-				catch (InsufficientBalanceException ex)
+				catch (NotEnoughFundsException ex)
 				{
-					Money needed = ex.Minimum - ex.Actual;
-					NotificationHelpers.Error($"Not enough coins selected. You need an estimated {needed.ToString(false, true)} BTC more to make this transaction.", "");
+					var messageBuilder = new StringBuilder("Not enough coins selected.");
+					if (ex.Missing is { })
```

Comment:
Missing is set by NBitcoin in the N-th iteration of CoinSelector. It is not known if the N-th iteration had taken the fee into a count of the number of inputs was finalized. So ex.Missing cannot be used for anything, here. 
The UI already showing some numbers for the users, like the selected amount, estimated fee. The notification is also a UI feature so it is OK to use the values taken from the UI.

Reply:
missing =  entered amount + estimated fee - selected amount